src/cdomains/apron/linearTwoVarEqualityDomain.apron.ml

```diff
@@ -0,0 +1,952 @@
+    | _ -> res
+
+  let bound_texpr d texpr1 = timing_wrap "bounds calculation" (bound_texpr d) texpr1
+end
+
+module D =
+struct
+  include Printable.Std
+  include ConvenienceOps (Mpqf)
+  include VarManagement 
+
+  module Bounds = ExpressionBounds
+
+  module Convert = SharedFunctions.Convert (V) (Bounds) (struct let allow_global = true end) (SharedFunctions.Tracked) 
+
+  type var = V.t
+
+  let name () = "lin2vareq"
+
+  let to_yojson _ = failwith "ToDo Implement in future"
+
+  let is_bot t = equal t (bot ())
+  let is_bot_env t = t.d = None
+
+  (*this shows "top" for a specific environment to enable the calculations. It is the identity of all equalities*)
+  let identity env = {d = Some (Array.init (Environment.size env) (fun i -> (Some i, Z.zero))); env = env}
+
+  (*Should never be called but implemented for completeness *)
+  let top () = {d = Some (EArray.empty()); env = empty_env}
+
+  (*is_top returns true for identity array and empty array *)
+  let is_top t = GobOption.exists EArray.is_top_array t.d
+
+  (* prints the current variable equalities with resolved variable names *)
+  let show varM =
+    let lookup i = Var.to_string (Environment.var_of_dim varM.env i) in
+    let show_var i tuple =
+      match tuple with
+      | (None, offset) -> "Variable " ^ string_of_int i ^ " named " ^ (lookup i) ^ " equals " ^ Z.to_string offset ^ "\n"
+      | (Some index, offset) -> "Variable " ^ string_of_int i ^ " named " ^ (lookup i) ^ " equals " ^ lookup index ^ " + " ^ Z.to_string offset ^ "\n"
+    in if is_top varM then "⊤\n" else 
+      match varM.d with
+      | None -> "⊥\n"
+      | Some arr -> if is_bot varM then "Bot \n" else Array.fold_left (fun acc elem -> acc ^ elem ) "" (Array.mapi show_var arr)  
+
+  let pretty () (x:t) = text (show x)
+  let printXml f x = BatPrintf.fprintf f "<value>\n<map>\n<key>\nequalities-array\n</key>\n<value>\n%s</value>\n<key>\nenv\n</key>\n<value>\n%s</value>\n</map>\n</value>\n" (XmlUtil.escape (Format.asprintf "%s" (show x) )) (XmlUtil.escape (Format.asprintf "%a" (Environment.print: Format.formatter -> Environment.t -> unit) (x.env)))
+
+  let meet t1 t2 =
+    let sup_env = Environment.lce t1.env t2.env in
+    let t1, t2 = change_d t1 sup_env true false, change_d t2 sup_env true false in
+    let subst_var ts x t = 
+      match !ts with
+      | None -> ()
+      | Some ts' ->
+        if Array.length ts' <> 0 then
+          for i = 0 to Array.length ts' - 1 do
+            match ts'.(i) with
+            | (None, _) -> ()
+            | (Some x', b') -> if x = x' then
+                (match t with 
+                 | (None, bt) -> ts'.(i) <- (None, Z.(b' + bt))
+                 | (Some xt, bt) -> ts'.(i) <- (Some xt, Z.(b' + bt)))
+          done
+    in
+    let add_conj ts t i = 
+      match !ts with
+      | None -> ()
+      | Some ts' ->
+        (match t with
+         | (None, b) -> 
+           (match ts'.(i) with
+            | (None, b') -> if b <> b' then ts := None;
```

Comment:
Instead of the polymorphic compare, please use the appropriate function from `Z`.